### PR TITLE
cpp: Add removeChannel, missing include, operator==

### PR DESCRIPTION
**Pull request by jtbandes**
A couple of API and build improvements.

### CODE REVIEW

File: cpp/foxglove-websocket/include/foxglove/websocket/server.hpp [hunk 1]
```diff
@@ -31,11 +32,17 @@ struct ChannelWithoutId {
   std::string encoding;
   std::string schemaName;
   std::string schema;
+
+  bool operator==(const ChannelWithoutId& other) const {
+    return topic == other.topic && encoding == other.encoding && schemaName == other.schemaName &&
```
Comment (KerstinKeller) [inline cpp/foxglove-websocket/include/foxglove/websocket/server.hpp:37]: just do a 
```
bool operator==(ChannelWithoutId const& rhs) const = default;
```
It should be C++11 and the compiler generates just that.

File: cpp/foxglove-websocket/include/foxglove/websocket/server.hpp [hunk 2]
```diff
@@ -48,6 +55,10 @@ struct Channel : ChannelWithoutId {
       {"schema", channel.schema},
     };
   }
+
+  bool operator==(const Channel& other) const {
+    return id == other.id && ChannelWithoutId::operator==(other);
```
Comment (KerstinKeller) [inline cpp/foxglove-websocket/include/foxglove/websocket/server.hpp:60]: Same as above?

File: cpp/foxglove-websocket/include/foxglove/websocket/server.hpp (hunk 1 above)
Comment (jtbandes) [reply cpp/foxglove-websocket/include/foxglove/websocket/server.hpp:37]: Hmm, seems this is a c++20 feature? never heard of it before :) https://en.cppreference.com/w/cpp/language/default_comparisons

Comment (KerstinKeller) [reply cpp/foxglove-websocket/include/foxglove/websocket/server.hpp:37]: I could have sworn that I used it before, but apparently not... too bad 😞

**Pull request merged**